### PR TITLE
use junit5

### PR DESCRIPTION
**Pull request by snopoke**
@dimagi/formplayer 
Make tests use Junit5

Junit4 has been removed removed from spring 2.4 (we're on 2.2). We will still have to include the dependency since commcare-core requires Junit4

I'm not sure if we can upgrade commcare-core to Junit5 since it's used by commcare-android which also uses Junit4 and can't be upgraded easily to 5 because of some of the test tools it makes use of.

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/application/WebAppContext.java [hunk 1]
```diff
@@ -323,4 +324,8 @@ public LockProvider lockProvider(DataSource dataSource) {
                         .build()
         );
     }
+
+    @Bean public RequestContextListener requestContextListener(){
```
Comment (snopoke) [inline src/main/java/org/commcare/formplayer/application/WebAppContext.java:328]: this is the recommended way to make the request object available.

The only changes to non-test code are related to this - accessing the request from the `RequestContextHolder`

File: src/main/java/org/commcare/formplayer/util/RequestUtils.java [hunk 2]
```diff
@@ -78,4 +81,12 @@ public static String getHmac(String key, String data) throws Exception {
         sha256_HMAC.init(secret_key);
         return Base64.encodeBase64String(sha256_HMAC.doFinal(data.getBytes("UTF-8")));
     }
+
+    public static FormplayerHttpRequest getCurrentRequest() {
+        RequestAttributes attributes = RequestContextHolder.getRequestAttributes();
+        if (RequestContextHolder.getRequestAttributes() != null) {
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/util/RequestUtils.java:87]: nit: can reuse `attributes`

**Review by shubham1g5 — APPROVED** [review]
looks great.

File: src/main/java/org/commcare/formplayer/util/RequestUtils.java (hunk 2 above)
Comment (snopoke) [reply src/main/java/org/commcare/formplayer/util/RequestUtils.java:87]:
```suggestion
        if (attributes != null) {
```

**Pull request merged**